### PR TITLE
Fixing grass not generating below trees

### LINKED ISSUE #5135: Grass is not generated below trees
Client version: x.x.x
Server OS: Windows/Linux/macOS/BSD/...
Cuberite Commit id: (Found at the top of the server's console output)

### Expected behavior
Grass below trees

### Actual behavior
No grass

I think this is related that the height map includes trees but the generator checks for the block at the top of the height map which is in case of a tree not grass so doesn't plant grass

### PR DESCRIPTION
**Pull request by 12xx12**
Fixes #5135

### CODE REVIEW

File: src/Generating/FinishGen.cpp [hunk 1]
```diff
@@ -542,6 +542,21 @@ void cFinishGenTallGrass::GenFinish(cChunkDesc & a_ChunkDesc)
...
+			while (
+				(BlockBelow == E_BLOCK_LEAVES) ||
+				(BlockBelow == E_BLOCK_NEW_LEAVES) ||
+				(BlockBelow == E_BLOCK_LOG) ||
+				(BlockBelow == E_BLOCK_NEW_LOG) ||
+				(BlockBelow == E_BLOCK_AIR)
+			)
+			{
+				y--;
+				BlockBelow = a_ChunkDesc.GetBlockType(x, y - 1, z);
```
Comment (bearbin) [inline src/Generating/FinishGen.cpp:557]: This needs a bounds check, at least.

**Review by bearbin — CHANGES_REQUESTED** [review]

File: src/Generating/FinishGen.cpp (hunk 1 above)
Comment (12xx12) [reply src/Generating/FinishGen.cpp:557]: yeah - you're right

**Review by bearbin — APPROVED** [review]

**Comment by tigerw** [comment]
can we re-order the finishers so the grass generator runs before trees are planted?

**Comment by 12xx12** [comment]
that would work too

**Comment by bearbin** [comment]
I can imagine there would be weird edge cases doing it the other way too, any particular reason not to do it this way @tigerw?

**Comment by 12xx12** [comment]
Negative:
I am not sure what happens if the generator tries to place a tree where already grass is

Positive:
Less performance impact

**Comment by tigerw** [comment]
Performance mainly, ideally we'd reduce Y-major searching because the stride is 256 elements.

**Comment by bearbin** [comment]
I suggest we merge this unless somebody wants to do a performance test then.

**Comment by tigerw** [comment]
OK, I can investigate reordering later then.

**Pull request merged**